apps/api/src/mass-email-record/mass-email-record.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { PromisePoolError } from '@supercharge/promise-pool';
+import { AppLogger } from '../common/logger.service';
+import { MassEmailRecordEntity } from './entity/mass-email-record.entity';
+import { GenericError } from 'src/common/generic-exception';
+
+@Injectable()
+export class MassEmailRecordService {
+  constructor(
+    @InjectRepository(MassEmailRecordEntity)
+    private readonly massEmailRecordRepository: Repository<MassEmailRecordEntity>,
+    @Inject(Logger)
+    private readonly logger: AppLogger,
+  ) {}
+
+  async createMassEmailRecord(record: Partial<MassEmailRecordEntity>) {
+    const recordEntity = this.massEmailRecordRepository.create(record);
+    const result = await this.massEmailRecordRepository.save(recordEntity);
+    this.logger.log(`mass email record created: ${result.id}`);
+  }
+
+  /**
+   * maps data to a record entry for saving
+   *
+   * @param userId: requesting users id
+   * @param emailIds: list of registrant ids for corresponding emails
+   * @param errors: error object returned from PromisePool library
+   * @returns the created record
+   */
+  mapRecordObject(
+    userId: string,
+    emailIds: string[],
+    errors: PromisePoolError<{ status: string; name: string; message: string; userId: string }>[],
+  ): Partial<MassEmailRecordEntity> {
+    let mappedErrors: GenericError[] = [];
+    // general exceptions and AWS errors have different formats
+    // check for general exceptions then look for AWS errors
+    if (errors) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      mappedErrors = errors.map(({ error, recipientId }: any) => {
```

Comment:
Nit maybe use unknown instead of any here?

Reply:
I'd still need to do some form of type checking with unknown,  I will take this and look into it